migrate/20240121_move_fw_rules.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  up do
+    create_table(:firewall) do
+      column :id, :uuid, primary_key: true
+      column :name, :text, null: false, default: "Default"
+      column :description, :text, null: false, default: "Default firewall"
+      column :created_at, :timestamp, null: false, default: Sequel::CURRENT_TIMESTAMP
+      foreign_key :vm_id, :vm, type: :uuid, null: true
+    end
+
+    alter_table(:firewall_rule) do
+      add_foreign_key :firewall_id, :firewall, null: true, type: :uuid
+      set_column_allow_null :private_subnet_id
+      add_unique_constraint [:ip, :port_range, :firewall_id]
+    end
+
+    run <<~SQL
+INSERT INTO firewall (id, name, description, vm_id)
+SELECT id, 'Default', 'Default firewall', id FROM vm;
+    SQL
+
+    # populate firewall rules per vm
+    run <<~SQL
+WITH vm_subnet_mapping AS (
+  SELECT
+      vm.id AS vm_id,
+      nic.private_subnet_id
+  FROM
+      vm
+  INNER JOIN
+      nic ON vm.id = nic.vm_id
+),
+new_firewall_rules AS (
+    SELECT
+        gen_random_uuid() AS id, -- Generate new UUIDs for the id
```

Comment:
does this ubidify properly? and does that matter? (whether it's type_etc or something else). Worse yet, it could wrong-ubidfy by choosing the wrong prefix by chance. This is an interesting danger we (okay I, by requesting it) introduced into our system.

Reply:
No it doesn't. Here is an example generated ubid;
```
=> #<FirewallRule @values={:id=>"edce4855-572b-4a93-831b-49aa0dc7dae4", :ip=>"0.0.0.0/0", :port_range=>nil, :firewall_id=>"b5294c13-5f2b-8374-9669-f178c3057891"}>
[3] clover-development(main)> fw.ubid
=> "mkxq74gnaq5d4gcdmkagdrzde9"
[5] clover-development(main)> FirewallRule.from_ubid("mkxq74gnaq5d4gcdmkagdrzde9")
=> #<FirewallRule @values={:id=>"edce4855-572b-4a93-831b-49aa0dc7dae4", :ip=>"0.0.0.0/0", :port_range=>nil, :firewall_id=>"b5294c13-5f2b-8374-9669-f178c3057891"}>
```
I am not sure how important it is to try to eliminate the danger of picking a wrong prefix. 10 bits must line up perfectly for a minimal amount of firewall_rules. Furthermore, the ubid property of firewall_rules are only used in the UI component for now. 99% of our firewall_rules will not be using the ubid property at the time of deployment and they will be destroyed shortly.

Reply:
wait, but don't you persist these possibly-bad ubids forever? Maybe I am understanding. Sure, momentary danger-ish, not so big a deal if it makes the transition cheaper at our phase.

Reply:
answered it in here https://github.com/ubicloud/ubicloud/pull/1119#discussion_r1466631591